### ISSUE TITLE
dispatch: no capability gate on /v1/chat/completions — a chat request naming an embed/rerank-named model gets the raw upstream 500

### ISSUE THREAD

**Issue by thinmintdev** [Hal0ai/hal0#1894]
**Severity:** minor (enhancement) · **Box:** ct151-cpu-fresh.

### Summary

With the `embed` slot loaded, `POST /v1/chat/completions` naming any model whose id contains `embed` or `rerank` is forwarded to the embeddings llama-server (Rule 7 of capability routing, `_capability_resolve.py` `_EMBED_NAME_HINTS`) and the client gets that server's verbatim `HTTP 500 "the current context does not logits computation. skipping"`. `Dispatcher.forward()` passes upstream non-2xx bodies through verbatim by documented contract, and hal0 has no capability/type gate at dispatch time (`model_fit.slot_type_mismatch` only checks at bind time). With `embed` offline (the fresh-install default) the same request instead returns a clean typed 404.

This routing behavior itself is by-design (see `known-issues.yaml: chat-to-embed-slot-500-passthrough`) — the routing and the upstream passthrough are both intentional, test-locked designs. The gap is the **missing typed error**: hal0 should not leak a raw, confusing upstream 500 for a capability mismatch it can already detect from `/v1/models` labels.

### Repro

```
# With embed slot loaded:
curl -s -X POST $API/v1/chat/completions \
  -d '{"model":"<any model id containing embed or rerank>","messages":[{"role":"user","content":"hi"}]}'
# -> raw upstream 500, not a typed capability-mismatch error
```

### Expected

A typed `409`/`400` capability-mismatch envelope when a request names a model whose disclosed capability (`/v1/models` labels/recipe) does not include `chat`, instead of leaking the upstream's raw 500.

---
Found by `rc-validate` v1.0.0-rc.6. See `known-issues.yaml: chat-to-embed-slot-500-passthrough` for the full by-design adjudication of the routing/passthrough mechanism this wraps.
